msal/src/main/java/com/microsoft/identity/client/Account.java

```diff
@@ -93,6 +96,24 @@ public String getId() {
             id = mHomeOid;
         }
 
+        if (StringUtil.isEmpty(id)) {
+            com.microsoft.identity.common.logging.Logger.warn(
+                    TAG + methodName,
+                    "Unable to get account id from either ClientInfo or IdToken. Attempting to obtain from home account id."
+            );
+            id = AccountUtil.getUIdFromHomeAccountId(mHomeAccountId);
+        }
+
+        if (StringUtil.isEmpty(id)) {
+            com.microsoft.identity.common.logging.Logger.error(
+                    TAG + methodName,
+                    "Account ID is empty. Returning MISSING_FROM_THE_TOKEN_RESPONSE.",
+                    new NullPointerException("Account id is null or empty")
```

Comment:
you could pass null for this exception, IIRC... ;-)

Reply:
Done: https://github.com/AzureAD/microsoft-authentication-library-for-android/pull/1558/commits/6a560eacd0d8ba33606dec559aa3f03ffe8958fb